packages/firebase_core/firebase_core_platform_interface/lib/src/firebase_app_data.dart

```diff
@@ -0,0 +1,31 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:quiver_hashcode/hashcode.dart';
+
+import 'firebase_options.dart';
+
+/// A data class storing the name and options of a Firebase app.
+class FirebaseAppData {
```

Comment:
This relates to https://github.com/FirebaseExtended/flutterfire/pull/1392 which @kroikie just landed.

Generally in Flutterfire we try to match the native Firebase APIs as closely as we can while remaining idiomatic to the target platform. There's no equivalent to the FirebaseAppData class in the [native Firebase API](https://firebase.google.com/docs/reference/ios/firebasecore/api/reference/Classes/FIRApp#options
). There's just `FirebaseApp` and it exposes a `Future<FirebaseOptions> options` getter.

The reason why we have asynchronous `options` on firebase_core's `FirebaseApp` is that we want to allow apps to be configured on either the platform implementation side or through the `configure` API. But perhaps it would be better to make it synchronous as you've done here.

It seems a bit odd to have a class `FirebaseOptions` that matches the naming in the app-facing API exactly, but then we have another class `FirebaseAppData` that serves a similar function to `FirebaseApp` but has a slightly different class behavior. There are other differences that may be confusing, e.g. `Future<void> configure` versus `Future<FirebaseApp> configure`.

Because breaking changes to the platform interface are discouraged I want to make sure we're happy with the approach here. Some possible alternative paths:

1. Use Dart namespaces to disambiguate app-facing classes and platform interface classes and call this `FirebaseApp` but not require it to match firebase_core's `FirebaseApp`. The same could apply to `FirebaseOptions`. This limits code reuse and may be confusing to have two identically named classes that are subtly different.
1. Write the platform interface class in a way that can be exported and reused (so firebase_core and firebase_core_platform_interface share the same `FirebaseApp` and `FirebaseOptions` class).  We could either have options be `null` for FirebaseApp instances that were created using the direct constructor (which @kroikie exposed in the PR linked above to allow [cloud_firestore to do multiple app support in the codec](https://github.com/FirebaseExtended/flutterfire/blob/master/packages/cloud_firestore/lib/src/firestore_message_codec.dart#L91)) or store previously created apps in a `Map` and look them up as a factory constructor. This approach would makes it harder to make breaking changes to firebase_core.
1. We could have a convention that the platform interface class names are visually distinguishable from the class name in the the app-facing package (e.g. `FirebaseApp` and `PlatformFirebaseApp`). This might make it easier to understand that there are two similar but not identical classes.